content/src/content/jcr_root/apps/core/wcm/components/commons/datalayer/clientlibs/js/datalayer.js

```diff
@@ -38,13 +38,7 @@
         if (parentElement) {
 
             if (parentElement.tagName === "BODY") {
-                parentData = dataLayer.find(function(element) {
-                    return element.data !== undefined && element.data.page !== undefined;
-                });
-
-                if (parentData !== undefined) {
-                    component[componentID].parentId = Object.keys(parentData.data.page)[0];
-                }
+                component[componentID].parentId = Object.keys(dataLayer.getState("page"))[0];
```

Comment:
I would get the page ID directly:
`component[componentID].parentId = parentElement.id`

Reply:
Then it probably makes no sense to tread `body` elements differently, we could use `component[componentID].parentId = parentElement.id` for both cases, right?

Reply:
yes, makes sense.

Reply:
Done in 8695193